services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -145,3 +145,22 @@ def is_graph_equal(lhs_workbench: Dict, rhs_workbench: Dict) -> bool:
     except (ValueError, TypeError, AttributeError):
         return False
     return True
+
+
+async def project_uses_available_services(
+    project: Dict, available_services: List[Dict]
+) -> bool:
+    if not project["workbench"]:
+        # empty project
+        return True
+    # get project services
+    needed_services: Set[Tuple[str, str]] = {
+        (s["key"], s["version"]) for _, s in project["workbench"].items()
+    }
+
+    # get available services
+    available_services: Set[Tuple[str, str]] = {
+        (s["key"], s["version"]) for s in available_services
+    }
+
+    return needed_services.issubset(available_services)
```

Comment:
👍

Reply:
when you say forever. you talk devel mode or production?
also this would go faster with a proper solution using a separate table in the database. I will create an issue for doing this later.

Reply:
I refer to develop mode. I had a noticeable slow down.

If these permissions could be precomputed and cached, the user will have a better time while opening/accessing their stuff. I think a lot of systems in the platform will benefit from something like this.

Reply:
yes I agree that this is not ideal in terms of performance. but the list of available services for a user changes over time (at any time actually) so caching is difficult. 
I will try if I find some other improvement options

Reply:
ok so actually found a faster way. since I only need to filter using key/version of the services I only request these from the catalog, which then does not need to call the director as well. thus speed is back to something reasonable.

Reply:
great!